feat: add edit to payment builder action

CzarekDryl opened the pull request:
## Description

This PR adds the ability to edit actions after the locked state during the Funding, Release steps. Editing in Payment step is not possible for now.

- Edit Mode:
  - Displays yellow border and Edit Pill.
  - Disables stepper buttons and hides Kebab menu.
  - Makes Recipients, Amounts, and Claim Delays editable.
  - Change payment button opens modal, disabled until changes are made.

- Change Payment Modal:
  - Dropdown for execution options (Reputation and Permissions).

- Changes Step in Stepper:
  - Changes pill appears after edits in Funding, Release step.
  - Shows details on edits, permissions, and a change counter.

- If there is funding needed after editing, new pill should show after changes one

## Testing

* Install Lazy Consensus extension to test funding with motion
* Create advanced payment action and lock it
* Play around with funding and editing

[Design](https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-v5?type=design&node-id=4102-376576&mode=design&t=lVqi0Mno6dqgrCs8-4)

Resolves [https://github.com/JoinColony/colonyCDapp/issues/2238](https://github.com/JoinColony/colonyCDapp/issues/2238)

arrenv reviewed (changes requested):
@CzarekDryl Nice one, a lot of work here, thank you for tackling it. I spent some time testing and came across a few things:

1. Change payment modal is showing two rows of buttons as well as a red button. There should only be the Confirm changes button and it should not be red.
![image](https://github.com/user-attachments/assets/e5de076e-5b7a-49a2-bc7c-5d19c5c40f2f)

2. Clicking on the "Confirm changes" button did not do anything other than close the modal.

3. Making a change in the Review stage showed the changes step after it, which I was not able to click on. While in the Review stage, the changes pill should appear before review and be clickable so that you can see who made changes, when and how (i.e. as the Payment creator), and be able to handle multiple.
![image](https://github.com/user-attachments/assets/de0212f5-0212-4911-9d66-fb7491e2f124)

4. There are some formatting issues with the edit mode, the recipients change to being unverified and the inputs change width, also, is there anyway to make the yellow border coming in not impact the content width?

    #### Read only
    ![image](https://github.com/user-attachments/assets/882d67d3-09d9-4774-a676-03c05887aab4)

    #### Edit mode
    ![image](https://github.com/user-attachments/assets/e81237b9-6000-46ba-9aae-b5fb81c74999)

5. Confirm details button should be disabled while in edit mode, and while the edit transaction is happening in order to prevent a clash.
![image](https://github.com/user-attachments/assets/162a120c-d190-4ffe-9e4e-d8d208beb974)

6. It seems the changes count and change history is not working fully it seems to only count the amount changes, in the following screenshot I had edited the following, a total of 5 changes but it only shows 2 changes, changes included:
    - 1st Row: Amount was changed from 2 to 1, claim delay was changed from 1 to 0.
    - 2nd Row: Recipient was changed from amy to leela, amount was changed from 3 to 1, claim delay was changed from 2 to 0.
   
    ![image](https://github.com/user-attachments/assets/da6aa99a-5991-4c62-9aab-82c20032693a)

7. Trying to fund the payment crashed the app with the following error and I could not continue testing any further.
![image](https://github.com/user-attachments/assets/fad13071-9c93-4746-92f8-52d68054f934)

CzarekDryl commented:
@arrenv 

1. Fixed
2. Fixed
3. and 5. in this issue I'm taking care of editing after the locked state
4. Which browser are you testing this on? On my end, the gaps aren't that large — it works like when creating actions
<img width="660" alt="image" src="https://github.com/user-attachments/assets/cf4d5e5f-648d-4d25-a482-bb3ed2d8057c">

6. It's working fine for me
<img width="983" alt="image" src="https://github.com/user-attachments/assets/447c1b3c-9f98-4800-88f5-cdecfe7cffbc">
7. Fixed
<img width="369" alt="image" src="https://github.com/user-attachments/assets/5db2a6a8-a2b9-48f4-bceb-dfe7fdf5c57f">
